CleverSDK/Classes/CLVOAuthManager.m

```diff
@@ -93,8 +117,52 @@ + (NSString *)state {
 }
 
 + (void)login {
+    NSString *state = [self generateRandomString:32];
+    [self setState:state];
+    
+    NSString *safariURLString = [NSString stringWithFormat:@"https://clever.com/oauth/authorize?response_type=code&client_id=%@&redirect_uri=%@&state=%@",
+                                 [self clientId], [self redirectUri], [self state]];
+    
+    NSString *cleverAppURLString = [NSString stringWithFormat:@"com.clever://oauth/authorize?response_type=code&client_id=%@&redirect_uri=%@&state=%@&sdk_version=%@", [self clientId], [self redirectUri], [self state], SDK_VERSION];
+    
     CLVOAuthManager *manager = [self sharedManager];
-    [[manager clvLogin] login];
+    if (manager.districtId) {
+        NSString *districtId = manager.districtId;
+        if (districtId == nil) {
+            districtId = @"";
+        }
+        safariURLString = [NSString stringWithFormat:@"%@&district_id=%@", safariURLString, districtId];
+        cleverAppURLString = [NSString stringWithFormat:@"%@&district_id=%@", safariURLString, districtId];
+    }
+    
+    // iOS 8 - always use Safari. Clever App not supported
+    if (SYSTEM_VERSION_LESS_THAN(@"9.0")) {
+        [[UIApplication sharedApplication] openURL: [NSURL URLWithString:safariURLString]];
+        return;
+    }
+    
+    // Switch to native Clever app if possible
+    if ([[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:cleverAppURLString]]) {
+        [[UIApplication sharedApplication] openURL:[NSURL URLWithString:cleverAppURLString] options:@{} completionHandler:nil];
+        return;
+    }
+    
+    // Fallbacks:
+    // iOS 9/10 - use SFSafariViewController
+    // iOS 11+ - use Safari
+    if (SYSTEM_VERSION_LESS_THAN(@"11.0")) {
+        SFSafariViewController *svc = [[SFSafariViewController alloc] initWithURL:[NSURL URLWithString:safariURLString] entersReaderIfAvailable:NO];
+        if ([[self uiDelegate] presentedViewController]) {
+            [[self uiDelegate] dismissViewControllerAnimated:YES completion:^{
+                [[self uiDelegate] presentViewController:svc animated:YES completion:nil];
+            }];
+        } else {
+            [manager.uiDelegate presentViewController:svc animated:YES completion:nil];
```

Comment:
Should we do `[self uiDelegate]` to be consistent? I think they do the same thing, right?